src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -247,25 +258,31 @@ private boolean checkInstallationDir(Path path) throws NoSuchFileException, Acce
     }
     
     private void checkEnvironmentProperties() {
-        this.blackduckHost = System.getenv(SYS_ENV_PROP_BLACKDUCK_URL);
-        final String httpsHost = System.getenv(SYS_ENV_PROP_PROXY_HTTPS_HOST);
-        final String httpHost = System.getenv(SYS_ENV_PROP_PROXY_HTTP_HOST);
+        this.blackduckHost = utility.getSysEnvProperty(SYS_ENV_PROP_BLACKDUCK_URL);
+        final String httpsHost = utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_HOST);
+        final String httpHost = utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTP_HOST);
         
         if (httpsHost != null && StringUtils.isNotBlank(httpsHost)) {
             /* HTTPS Proxy System Environment settings are preferred. */
-            proxyProperties.put(ARG_PROXY_PORT, httpsHost);
-            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTPS_PORT));
-            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTPS_USERNAME));
-            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTPS_PASSWORD));
+            proxyPropertiesPut(ARG_PROXY_HOST, httpsHost);
+            proxyPropertiesPut(ARG_PROXY_PORT, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_PORT));
+            proxyPropertiesPut(ARG_PROXY_USERNAME, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_USERNAME));
+            proxyPropertiesPut(ARG_PROXY_PASSWORD, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_PASSWORD));
```

Comment:
The wrapper method proxyPropertiesPut was added to put value into the propertiesMap only if the value is not a null.